ui/src/geoviite-design-lib/track-meter/track-meter.tsx

```diff
@@ -1,14 +1,39 @@
-import { formatTrackMeter } from 'utils/geography-utils';
+import { formatTrackMeter, formatTrackMeterWithoutMeters } from 'utils/geography-utils';
 import * as React from 'react';
 import { TrackMeter } from 'common/common-model';
+import { Point } from 'model/geometry';
+
+type DisplayedAddressPoint = {
+    point: Point;
+    address: TrackMeter;
+};
 
 type TrackMeterProps = {
-    value?: TrackMeter;
+    addressPoint?: DisplayedAddressPoint;
     placeholder?: string;
+    onShowOnMap?: () => void;
+    displayDecimals?: boolean;
 };
 
-const TrackMeter: React.FC<TrackMeterProps> = ({ value, placeholder }: TrackMeterProps) => {
-    return <React.Fragment>{value ? formatTrackMeter(value) : placeholder ?? ''}</React.Fragment>;
+const TrackMeter: React.FC<TrackMeterProps> = ({
+    addressPoint,
+    placeholder,
+    onShowOnMap,
+    displayDecimals = true,
+}: TrackMeterProps) => {
+    if (!addressPoint || !onShowOnMap) {
+        return <React.Fragment>{placeholder ?? ''}</React.Fragment>;
+    }
+
+    const displayedTrackMeterValue = displayDecimals
+        ? formatTrackMeter(addressPoint.address)
+        : formatTrackMeterWithoutMeters(addressPoint.address);
+
+    return (
+        <a className={'link'} onClick={onShowOnMap}>
+            {displayedTrackMeterValue}
+        </a>
```

Comment:
Pari huomiota:
* Vaikka muuten olenkin early returnien kannattaja, niin UI-komponenteissa sanoisin että yksittäinen return kaiken lopussa voisi olla parempi ratkaisu. Syynä lähinnä se, että UI-komponenteissa on yleensä merkittävästi logiikkaa returnien sisällä.
* Sanoisin myös että parametritasolla tälle komponentille voisi edelleen passata vaan `trackMeter`in koko `addressPoint`in sijasta, sillä tuolta `addressPoint`ista ei näytettäisi tarvivan täällä mitään muuta. 
* Näyttäisi siltä että linkkiyden lisäksi myös päättely siitä näytetäänkö annettu ratakilometri vai placeholder menee nyt myös `onShowOnMap`-parametrin perusteella. Meillä on harvinaisia tilanteita joissa halutaan näyttää ratakilometrejä ilman että ne ovat linkkejä (esim. reviewin seuraava tiedosto `location-holder-view.tsx`,) johon ratakilometri joudutaan nyt passaamaan placeholderissa.

Reply:
* Sinänsä juu, mutta samalla menetetään TS:n tyyppirajauksen hyötyjä -> tulee herkästi paljon sisennystä iffittelyistä. Vähän niin ja näin. 
* Truedat, tunkatessa komponentille päätyi alunperin kartalla näytettävän bounding-boxin määritys, joka kaipasi myös koordinaattia. Tämän muuttelen alkuperäiseksi yksittäisarvoksi.
* Korjaantuisi vastaavasti, jos tänne passattaisiin pelkkä näytettävä arvo, eikä komponentti handlaisi formatointia, joskin kutsuja päätyisi taas monimutkaisemmaksi. Hmm 🤔. Sinänsähän tuo `location-holder-view.tsx` on uniikki tapaus muutenkin, eli ehkä sen ei edes tätä komponenttia pitäisi hyödyntää.

Reply:
Totta tuo viimeinen pointti. Sinällään jos nuo linkittömät käyttötapaukset hoitaisi jotenkin kokonaan erillään (varsinkin jos tuo `location-holder-view.tsx` on koodipesän ainoa sellainen tapaus,) niin tältä komponentiltahan voisi pudottaa kokonaan tuen pelkän tekstin näyttämiseltä -> koko komponentti yksinkertaistuisi merkittävästi toteutuksen osalta ja sen voisi myös nimetä uudelleen `track-meter-link`:iksi. Toki huonona puolena se, että sen jälkeen komponentti tekee aina yli yhtä asiaa (näyttää ratakilometrisijainnin JA tekee siitä linkin,) mutta niin se tekisi nyt jo muutenkin